extra_tests/snippets/stdlib_struct.py

```
@@ -76,3 +76,19 @@ def __index__(self):
 assert struct.error.__module__ == "struct"
 assert struct.error.__name__ == "error"
+
+# Non-ASCII format string: error type matches CPython.
+# str → UnicodeEncodeError (encoding='ascii')
+# bytes → struct.error
+try:
+    struct.Struct("\udc00")
+except UnicodeEncodeError as e:
+    assert e.encoding == "ascii"
+else:
+    assert False, "expected UnicodeEncodeError"
```

Ruff (0.15.11)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)